src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -0,0 +1,43 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.CinemaHallDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.CinemaHall;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.query.Query;
+
+@Dao
+public class CinemaHallDaoImpl implements CinemaHallDao {
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            session.save(cinemaHall);
+            return cinemaHall;
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't save the cinemaHall " + cinemaHall, e);
+        }
+    }
+
+    @Override
+    public Optional<CinemaHall> get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return Optional.ofNullable(session.get(CinemaHall.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get the cinemaHall with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<CinemaHall> query = session.createQuery("FROM CinemaHall", CinemaHall.class);
+            return query.list();
```

Comment:
You can inline this string of code instead of the existing one. For instance, you can write 'return session.createQuery("from CinemaHall", CinemaHall.class).getResultList(). It will be a little simplified, won't it?